libs/langchain/langchain/llms/koboldai.py

```diff
@@ -122,77 +124,117 @@ class KoboldApiLLM(LLM):
 
         Args:
             prompt: The prompt to use for generation.
             stop: A list of strings to stop generation when encountered.
 
         Returns:
-            The generated text.
+            The generated text stripped of stop sequences and leading or trailing whitespace.
 
         Example:
             .. code-block:: python
 
-                from langchain.llms import KoboldApiLLM
-
-                llm = KoboldApiLLM(endpoint="http://localhost:5000")
-                llm("Write a story about dragons.")
+                from langchain.llms import koboldApiLLM
+                llm = koboldApiLLM(endpoint="http://localhost:5000")
+                llm("Write a story about llamas.")
         """
-        data: Dict[str, Any] = {
-            "prompt": prompt,
-            "use_story": self.use_story,
-            "use_authors_note": self.use_authors_note,
-            "use_world_info": self.use_world_info,
-            "use_memory": self.use_memory,
-            "max_context_length": self.max_context_length,
-            "max_length": self.max_length,
-            "rep_pen": self.rep_pen,
-            "rep_pen_range": self.rep_pen_range,
-            "rep_pen_slope": self.rep_pen_slope,
-            "temperature": self.temperature,
-            "tfs": self.tfs,
-            "top_a": self.top_a,
-            "top_p": self.top_p,
-            "top_k": self.top_k,
-            "typical": self.typical,
-        }
-
-        if stop is not None:
-            data["stop_sequence"] = stop
 
-        response = requests.post(
-            f"{clean_url(self.endpoint)}/api/v1/generate", json=data
-        )
+        url = f"{clean_url(self.endpoint)}/api/v1/generate"
+        params = self._get_parameters(stop)
+        request = params.copy()
+        request["prompt"] = prompt
 
-        response.raise_for_status()
-        json_response = response.json()
+        response = requests.post(url, json=request)
 
-        if (
-            "results" in json_response
-            and len(json_response["results"]) > 0
-            and "text" in json_response["results"][0]
-        ):
-            text = json_response["results"][0]["text"].strip()
+        if response.status_code == 200:
+            text = response.json()["results"][0]["text"]
+            stop_sequences = params["stop_sequence"]
 
-            if stop is not None:
-                for sequence in stop:
+            if stop_sequences:
+                for sequence in stop_sequences:
                     if text.endswith(sequence):
-                        text = text[: -len(sequence)].rstrip()
-
-            return text
+                        text = text.rsplit(sequence, 1)[0]
         else:
-            raise ValueError(
-                f"Unexpected response format from Kobold API:  {json_response}"
-            )
+            # If the response was not successful, print an error and set the text to an empty string
+            print(f"ERROR: response: {response}")
```

Comment:
are we sure we don't want to actually error here?

at the very least should we log the error instead of printing (logger.error...)

could even be worth making the LLM response itself an error message

Reply:
Good point how does this change look?

```python
error_message = f"Unexpected response format from Kobold API: {json_response}"
logger.error(error_message)
text = error_message
```


I should probably also change the `_get_parameters` error on line 184. I think we can skip the error when `stop` and `self.stop_sequence` are both set. `stop` can override `self.stop_sequence` when it's given:

```python
params["stop_sequence"] = stop if stop is not None else (self.stop_sequence or [])
```

What do you think?